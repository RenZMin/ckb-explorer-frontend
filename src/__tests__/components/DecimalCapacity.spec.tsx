import { ReactElement } from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import DecimalCapacity from '../../components/DecimalCapacity'
import { DecimalPartPanel, DecimalZerosPanel } from '../../components/DecimalCapacity/styled'

describe('DecimalCapacity Component', () => {
  let component: ReactElement

  beforeAll(() => {
    component = <DecimalCapacity value="128.1234" />
  })

  it('shallow renders', () => {
    const wrapper = renderer.create(component).toJSON()
    expect(wrapper).toMatchSnapshot()
  })

  it('Component Render', () => {
    const wrapper = shallow(component)
    expect(wrapper).toBeDefined()
    expect(wrapper.find(DecimalPartPanel)).toHaveLength(1)
    expect(wrapper.find(DecimalPartPanel).text()).toBe('.1234')
    expect(wrapper.find(DecimalZerosPanel)).toHaveLength(1)
    expect(wrapper.find(DecimalZerosPanel).text()).toBe('0000')
  })
})
