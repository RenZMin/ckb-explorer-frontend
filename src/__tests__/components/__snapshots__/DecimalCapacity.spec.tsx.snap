// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DecimalCapacity Component shallow renders 1`] = `
<div
  className="sc-bdnxRM dqBRPR"
>
  <span>
    128
  </span>
  <div
    className="sc-gtsrHT hYceg monospace"
  >
    .1234
  </div>
  <div
    className="sc-dlnjwi huabTw monospace"
  >
    0000
  </div>
  <div
    className="decimal__unit monospace"
  >
    CKB
  </div>
</div>
`;
